### PR TITLE
fix: [PL-31050]: Added a retry of 3 times when creating redis client

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
Added a re-try mech when creating a redis client as we are seeing an increased number of timeout exceptions when connecting to redis.
## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42936)

### CODE REVIEW

File: 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java [hunk 1]
```diff
@@ -125,11 +129,19 @@ public RedissonClient getClient(RedisConfig redisConfig) {
...
+      int failedAttempts = 0;
+      while (true) {
+        try {
+          redisConfigRedissonClientMap.put(redisConfig, Redisson.create(config));
+          break;
+        } catch (Exception ex) {
+          failedAttempts++;
+          log.warn("Exception occurred when creating redis client. Trail num: {}", failedAttempts, ex);
+          if (failedAttempts == NUM_OF_RETRIES) {
+            throw ex;
```
Comment (kshitij-harness) [inline 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java:141]: log an error here

File: 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java [hunk 2]
```diff
@@ -125,11 +129,19 @@ public RedissonClient getClient(RedisConfig redisConfig) {
...
+          }
+          sleep(ofMillis(1000));
```
Comment (kshitij-harness) [inline 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java:143]: Make 1000 as a constant

File: 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java [hunk 3]
```diff
@@ -33,6 +36,7 @@
 @Slf4j
 public class RedissonClientFactory {
   private static final int DEFAULT_MIN_CONNECTION_IDLE_SIZE = 5;
+  private final int NUM_OF_RETRIES = 3;
```
Comment (kshitij-harness) [inline 960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java:39]: this should be static

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42936)* status: 1 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java` line 39 at r1](https://reviewable.io/reviews/harness/harness-core/42936#-NMT9PR-ow9aqi0mgGEP-r1-39:-NMTDkS6D7Nbv62b5c5U:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/949338604a49fe5b26c38073de9529790e5912c3/960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java#L39)):*
<details><summary><i>Previously, kshitij-harness (Kshitij Rastogi) wrote…</i></summary><blockquote>

this should be static
</blockquote></details>

Done.
___
*[`960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java` line 141 at r1](https://reviewable.io/reviews/harness/harness-core/42936#-NMT9PR-ow9aqi0mgGEP-r1-141:-NMTDkkQ6ed4sAgmvN2d:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/949338604a49fe5b26c38073de9529790e5912c3/960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java#L141)):*
<details><summary><i>Previously, kshitij-harness (Kshitij Rastogi) wrote…</i></summary><blockquote>

log an error here
</blockquote></details>

Done.
___
*[`960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java` line 143 at r1](https://reviewable.io/reviews/harness/harness-core/42936#-NMT9PR-ow9aqi0mgGEP-r1-143:-NMTDl8lBIQTJk6IrlBv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/949338604a49fe5b26c38073de9529790e5912c3/960-persistence/src/main/java/io/harness/redis/RedissonClientFactory.java#L143)):*
<details><summary><i>Previously, kshitij-harness (Kshitij Rastogi) wrote…</i></summary><blockquote>

Make 1000 as a constant
</blockquote></details>

Done.

**Review by kshitij-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42936)* status: 1 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained.

**Review by kshitij-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42936)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by VikasMaddukuriHarness** [comment]
trigger messagecheck

**Pull request merged**